File: conversione_base.java
import java.util.ArrayList;
import java.util.Scanner;

@SuppressWarnings("unused")
public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.println("Numero da convertire");
    int num = sc.nextInt();
    System.out.println("Base da cui convertire");
    int base1 = sc.nextInt();
    System.out.println("Base a cui convertire");
    int base2 = sc.nextInt();

    System.out.println(STR."Il numero \{num} (base \{base1}) diventa \{base2base(num, base1, base2)} in base \{base2}");
}

private static int bto10(int number, int base) {
    final int tt = number;
    int result = 0;
    ArrayList<Integer> list = new ArrayList<>();
    while (number > 0) {
        final int temp = number % 10;
        list.add(temp);
        if (temp >= base)
            throw new IllegalArgumentException(STR."La cifra \{temp} in \{tt} è maggiore o uguale alla base \{base}");
        number /= 10;
    }
    for (int i = 0; i < list.size(); i++) {
        int k = list.get(i);
        result += (int) (k * Math.pow(base, i));
    }
    return result;
}

private static int tenToB(int number, int base) {
    final ArrayList<Integer> list = new ArrayList<>();
    do {
        list.add(number % base);
        number /= base;
    } while (number != 0);
    int result = 0;
    for (int i = 0; i < list.size(); i++) {
        result += (int) (list.get(i) * Math.pow(10, i));
    }
    return result;
}

private static int base2base(int number, int inBase, int outBase) {
    return tenToB(bto10(number, inBase), outBase);
}

File: conversione_base.cpp
#include <iostream>
#include <list>
#include <cmath>
#include <valarray>
#include <vector>
using namespace std;

int bTo10(int number, const int base) {
    const int tt = number;
    int result = 0;
    vector<int> list;
    while(number > 0) {
        const int temp = number % 10;
        list.push_back(temp);
        if(temp >= base)
            throw invalid_argument("La cifra " + to_string(temp) + " in " + to_string(tt) + " è maggiore o uguale alla base " + to_string(base));
        number /= 10;
    }
    for(int i = 0; i < list.size(); i++) {
        const int k = list.at(i);
        result += k * static_cast<int>(pow(base, i));
    }
    return result;
}

int tenToB(int number, int base) {
    vector<int> list;
    do {
        list.push_back(number % base);
        number /= base;
    } while (number != 0);
    int result = 0;
    for(int i = 0; i < list.size(); i++) {
        result += list.at(i) * static_cast<int>(pow(10, i));
    }
    return result;
}

int base2base(int number, int inBase, int outBase) {
    return tenToB(bTo10(number, inBase), outBase);
}


int main() {
    int num, base1, base2;
    cout << "Numero da convertire" << endl;
    cin >> num;
    cout << "Base da cui convertire" << endl;
    cin >> base1;
    cout << "Base a cui convertire" << endl;
    cin >> base2;

    cout << "Il numero " + to_string(num) + " (base " + to_string(base1) + ") diventa " + to_string(base2base(num, base1, base2)) + " in base " + to_string(base2) << endl;
    return 0;
}

